packages/lesswrong/server/migrations/20250501T183247.ultraFeedEventsIndexes.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Migration to add optimized indexes to UltraFeedEvents table
+ * 
+ * Indexes are optimized for queries in:
+ * - CommentsRepo.getCommentsForFeed
+ * - SpotlightsRepo.getUltraFeedSpotlights
+ * - PostsRepo.getLatestPostsForUltraFeed
+ */
+
+export const up = async ({db}: MigrationContext) => {
+  // Primary index for most queries - userId first as recommended
+  await db.none(`
+    CREATE INDEX IF NOT EXISTS "ultraFeedEvents_userId_collectionName_eventType_createdAt_idx" 
+    ON "UltraFeedEvents" ("userId", "collectionName", "eventType", "createdAt");
+  `);
+
+  // Secondary index for document lookups - also starts with userId
+  await db.none(`
+    CREATE INDEX IF NOT EXISTS "ultraFeedEvents_userId_collectionName_documentId_idx" 
+    ON "UltraFeedEvents" ("userId", "collectionName", "documentId");
+  `);
```

Comment:
We have an `updateIndexes` helper function; you don't need to write these out by hand.  It creates the SQL from the indexes defined in the `indexSet` on the collection.